lab_2/main.py

```diff
@@ -149,7 +237,27 @@ def calculate_distance_sparse(unknown_text_vector: list,
     :param unknown_text_vector: sparse vector for unknown text
     :param known_text_vector: sparse vector for known text
     """
-    pass
+    if not isinstance(unknown_text_vector, list) or not isinstance(known_text_vector, list):
+        return None
+    for number in unknown_text_vector:
+        if not isinstance(number, list):
+            return None
+    for number in known_text_vector:
+        if not isinstance(number, list):
+            return None
+
+    distance_sum = 0
+    distance_dict = {}
+    for pare in known_text_vector:
+        distance_dict[pare[0]] = pare[1]
+    for pare in unknown_text_vector:
+        if pare[0] in distance_dict.keys():
+            distance_dict[pare[0]] -= pare[1]
+        else:
+            distance_dict[pare[0]] = pare[1]
+    for value in distance_dict.values():
+        distance_sum += value ** 2
+    return round(distance_sum ** 0.5, 5)
```

Comment:
and here too?

Reply:
this function calculates distance for sparsed vecrtor. firstly it adds all pares index(key)-frequency(value) from known_text_vector to dictionary, then it stars to add pares from unknown_text_vector, but if pares from unknown and known vectors contain frequencies with same indexes, funtion adds index and difference between frequencies. then it calulates sum of squared values and extracts square root